File: data/src/test/java/com/tonitealive/app/data/repository/DefaultUsersRepositoryTest.java
package com.tonitealive.app.data.repository;

import com.tonitealive.app.data.net.ToniteAliveApi;

import org.junit.Before;
import org.junit.Rule;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;


public class DefaultUsersRepositoryTest {

    @Rule
    public final MockitoRule mockitoRule = MockitoJUnit.rule();

    @Mock private ToniteAliveApi mockApi;

    private DefaultUsersRepository usersRepository;

    @Before
    void setup() {
        usersRepository = new DefaultUsersRepository(mockApi);
    }

}

File: data/src/test/kotlin/com/tonitealive/app/data/repository/DefaultUsersRepositoryTest.kt
package com.tonitealive.app.data.repository

import com.tonitealive.app.data.net.ToniteAliveApi
import org.junit.Before
import org.junit.Rule
import org.mockito.Mock
import org.mockito.junit.MockitoJUnit


class DefaultUsersRepositoryTest {

    @get:Rule
    val mockitoRule = MockitoJUnit.rule()

    @Mock lateinit var mockApi: ToniteAliveApi

    lateinit var usersRepository: DefaultUsersRepository

    @Before
    fun setup() {
        usersRepository = DefaultUsersRepository(mockApi)
    }

}